mjh_sanity/godas_obssanitycheck.py

```diff
@@ -0,0 +1,167 @@
+        formatter = mdates.DateFormatter('%Y/%m/%d')
+
+        fig = plt.figure(tight_layout=True)
+        gs = gridspec.GridSpec(2, 2)
+        ax = fig.add_subplot(gs[0, 0])
+        ax.plot_date(avail_dates, nlocs,'-og')
+        ax.title.set_text('No of obs available on each day')
+        ax.set_ylabel('No of obs')
+        #ax.set_xlabel('Dates')
+        ax.xaxis.set_major_locator(mdates.DayLocator(interval=10))
+        #ax.xaxis.set_tick_params(rotation=45, labelsize=9)
+        ax.set_xticklabels([])
+        #fig, ax = plt.subplots()
+        ax = fig.add_subplot(gs[1, 0])
+        plt.title('Min and Max value on each day')
+        plt.plot_date(avail_dates, minval,'-og', label='Min')
+        plt.plot_date(avail_dates, maxval,'-*r', label='Max')
+        ax.set_ylabel('Min/Max')
+        #ax.set_xlabel('Dates')
+        ax.legend(loc=0)
+        ax.xaxis.set_major_locator(mdates.DayLocator(interval=10))
+        #ax.xaxis.set_tick_params(rotation=40, labelsize=9)
+        ax.set_xticklabels([])
+        ax=fig.add_subplot(gs[:,1])
+        plt.title('missing files on dates:')
+        md_len=len(missing_dates)
+        print('number of missing files', md_len)
+        if md_len == 0:
+            plt.text(.12, .75, 'No missing files!', fontsize=13)
+            plt.text(.06, .65, 'from %s to %s'%(avail_dates[0], avail_dates[-1]))
+        else:
+            xstep=0.01
+            ystep=0.95
+            for tx in missing_dates:
+                plt.text(xstep, ystep, '%s'%tx, fontsize=8)
+                ystep=ystep-0.05
+                if ystep < .005: 
+                    xstep=xstep+0.33
+                    ystep=0.95
+        
+        ax.set_xticklabels([])
+        ax.set_yticklabels([])
+        plt.suptitle("File descriptor: %s"%file_descriptor, y=.95);
+        plt.savefig('Fig_%s_%s.png'%(file_descriptor, self.year), bbox_inches='tight', pad_inches=0.1)
+
+
+    def find_nlocs(self, filename):
+        args=self.args
+        nc_fid = Dataset(filename,'r')
+        nloc=nc_fid.variables['nlocs'].size
+        nc_fid.close()
+        return nloc
+
+    def find_minmax(self, fname):
+        args=self.args
+        ncfile = Dataset(fname,'r')
+        metadata_grp=ncfile.groups['MetaData']
+        var_grp=ncfile.groups[args.group]
+        var_tmp=np.squeeze(var_grp.variables[args.variable][:])
+        minval=min(var_tmp)
+        maxval=max(var_tmp)
+        ncfile.close()
+        return minval, maxval
+        
+if __name__ == '__main__':
+    description = """ Ex: godas_obssanitycheck.py -f adt_c2 -o missing -g ObsValue 
+                                       -v sea_surface_temperature -p /path/to/folder
+                  """
+    # Command line argument
+    parser = ArgumentParser(
+        description=description,
+        formatter_class=ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '-f',
+        '--file_descriptor',
+        type=str, nargs='+',required=True)
+    parser.add_argument(
```

Comment:
might be ok as a first go, but I'd rather not have to provide a variable. You could get that info from the file without asking the user the variable name.